torch/_tensor_docs.py

```diff
@@ -1930,6 +1930,66 @@ def add_docstr_all(method, docstr):
 Out-place version of :meth:`~Tensor.index_put_`.
 """)
 
+add_docstr_all('index_reduce_',
+               r"""
+index_reduce_(dim, index, source, reduce, *, include_self=True) -> Tensor
+
+Accumulate the elements of ``source`` into the :attr:`self`
```

Comment:
Do you think it could help to add a mathematical formula here that describes an element of the output in terms of the inputs?

Further, a small Python snippet that explains this in terms of plain, but inefficient operators could be helpful too.

Reply:
Added a code snippet for it like we have for scatter, also added a similar snippet to index_add. Does it make more sense now?